virttest/utils_test/libvirt.py

```diff
@@ -1176,7 +1176,16 @@ def do_migration(self, vms, srcuri, desturi, migration_type, options=None,
                 eclipse_time = 0
                 if func:
                     stime = int(time.time())
-                    func(func_params)
+                    if func == process.run:
+                        try:
+                            if args['shell']:
+                                func(args['func_params'], shell=True)
+                            else:
+                                func(args['func_params'])
```

Comment:
I think no need to check the value of args['shell'].
func(args['func_params'], shell=args['shell'])  is enough and looks simple.